### ISSUE TITLE
C++ operator bool class member breaks linking

### ISSUE THREAD

**Issue by Twinki14** [matusnovak/doxybook2#14]
```
## Public Functions

|                | Name           |
| -------------- | -------------- |
|  | **[Box](/api/classes/class_box.html#function-box)**()  |
|  | **[Box](/api/classes/class_box.html#function-box)**(std::int32_t X, std::int32_t Y, std::int32_t Width, std::int32_t Height)  |
|  | **[Box](/api/classes/class_box.html#function-box)**(const [Box](/api/classes/class_box.html) & B)  |
| bool | **[Contains](/api/classes/class_box.html#function-contains)**(const [Point](/api/classes/class_point.html) & P) const  |
| bool | **[Contains](/api/classes/class_box.html#function-contains)**(const [Box](/api/classes/class_box.html) & B) const  |
| bool | **[Contains](/api/classes/class_box.html#function-contains)**(std::int32_t X, std::int32_t Y) const  |
| [Point](/api/classes/class_point.html) | **[GetMiddle](/api/classes/class_box.html#function-getmiddle)**() const  |
| std::int32_t | **[GetX2](/api/classes/class_box.html#function-getx2)**() const  |
| std::int32_t | **[GetY2](/api/classes/class_box.html#function-gety2)**() const  |
| bool | **[OnScreen](/api/classes/class_box.html#function-onscreen)**() const  |
| bool | **[IsValid](/api/classes/class_box.html#function-isvalid)**() const  |
| [Point](/api/classes/class_point.html) | **[GetUniformRandomPoint](/api/classes/class_box.html#function-getuniformrandompoint)**() const  |
| [Point](/api/classes/class_point.html) | **[GetBinomialRandomPoint](/api/classes/class_box.html#function-getbinomialrandompoint)**(double ProbabilityX, double ProbabilityY) const  |
| [Point](/api/classes/class_point.html) | **[GetNormalRandomPoint](/api/classes/class_box.html#function-getnormalrandompoint)**(double PercentageDeviationX, double PercentageDeviationY) const  |
| [Point](/api/classes/class_point.html) | **[GetHybridRandomPoint](/api/classes/class_box.html#function-gethybridrandompoint)**(double ProbabilityX, double ProbabilityY, double StandardDeviationX, double StandardDeviationY, double PointGenAccuracy) const  |
| [Point](/api/classes/class_point.html) | **[GetProfileHybridRandomPoint](/api/classes/class_box.html#function-getprofilehybridrandompoint)**() const  |
| [Point](/api/classes/class_point.html) | **[GetProfileHybridRandomPoint](/api/classes/class_box.html#function-getprofilehybridrandompoint)**(double PointGenAccuracy) const  |
| bool | **[operator==](/api/classes/class_box.html#function-operator==)**(const [Box](/api/classes/class_box.html) & B) const  |
| bool | **[operator!=](/api/classes/class_box.html#function-operator!=)**(const [Box](/api/classes/class_box.html) & B) const  |
|  | **[operator bool](/api/classes/class_box.html#function-operator bool)**() const  |
|  | **[~Box](/api/classes/class_box.html#function-~box)**()  |
```

Specifically
```
|  | **[operator bool](/api/classes/class_box.html#function-operator bool)**() const  |
``` 

Doxygen doesn't seem to recognize the return type like other operators,

(class xml generated by doxygen)
```xml
      <memberdef kind="function" id="class_box_1acbbf88d0829393ec25644ffe7475d060" prot="public" static="no" const="yes" explicit="no" inline="no" virt="non-virtual">
        <type>bool</type>
        <definition>bool Box::operator!=</definition>
        <argsstring>(const Box &amp;B) const</argsstring>
        <name>operator!=</name>
        <param>
          <type>const <ref refid="class_box" kindref="compound">Box</ref> &amp;</type>
          <declname>B</declname>
        </param>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="Core/Types/Box.hpp" line="49" column="14"/>
      </memberdef>
      <memberdef kind="function" id="class_box_1ad051f61bcb0be65272632ee39578e3a3" prot="public" static="no" const="yes" explicit="no" inline="no" virt="non-virtual">
        <type></type>
        <definition>Box::operator bool</definition>
        <argsstring>() const</argsstring>
        <name>operator bool</name>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
<para><simplesect kind="return"><para>True if the x and y axis coordinates are greater than or equal to zero, and the width and height are greater than zero </para>
</simplesect>
</para>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="Core/Types/Box.hpp" line="52" column="9"/>
      </memberdef>
```

I'm not sure if the cause is specifically related to the type not being reconignized, but the issue is the link created for operator bool is broken, instead of ```#function-operator bool``` it should be ```#function-operator-bool```

EDIT: Doxygen 1.8.18

**Comment by matusnovak** [comment]
Hi @Twinki14 

The anchor links are created by Doxybook2 based on the `<name>operator bool</name>` from Doxygen's output. It's a Doxybook2 fault.

The problem is that all static site generators create the `##` IDs differently. For example `## function operator bool` creates `<h2 id="function-operator-bool">...</h2>` and some other ones can create `<h2 id="functionoperatorbool">...</h2>`. It's difficult to create a link with anchor to match it precisely. 

I'll get back to this issue later this week, I just have other stuff at the moment.

**Comment by matusnovak** [comment]
Sorry it took me so long.

For now I have just replaced all spaces with dashes for the anchor part of the link. It seems to work in most cases. I have released it as [releases/tag/v1.1.5](https://github.com/matusnovak/doxybook2/releases/tag/v1.1.5). It's good enough for now but more work needs to be done (maybe I should add this behavior in the config file).
